internal/ls/findallreferences.go

```diff
@@ -0,0 +1,1486 @@
+package ls
+
+import (
+	"cmp"
+	"fmt"
+	"slices"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/microsoft/typescript-go/internal/ast"
+	"github.com/microsoft/typescript-go/internal/astnav"
+	"github.com/microsoft/typescript-go/internal/binder"
+	"github.com/microsoft/typescript-go/internal/checker"
+	"github.com/microsoft/typescript-go/internal/compiler"
+	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/lsp/lsproto"
+	"github.com/microsoft/typescript-go/internal/scanner"
+
+	"github.com/microsoft/typescript-go/internal/tspath"
+)
+
+// === types for settings ===
+type referenceUse int
+
+const (
+	referenceUseNone       referenceUse = 0
+	referenceUseOther      referenceUse = 1
+	referenceUseReferences referenceUse = 2
+	referenceUseRename     referenceUse = 3
+)
+
+type refOptions struct {
+	findInStrings                       bool
+	findInComments                      bool
+	use                                 referenceUse // other, references, rename
+	implementations                     bool
+	providePrefixAndSuffixTextForRename bool
```

Comment:
Can we take this opportunity to rename one of the worst and most confusingly named user preferences we have? This is going to be implemented differently too (rename just returns a generic workspace edit, no explicit prefix and suffix text properties), becoming even more of a misnomer. This is what it maps to:

<img width="574" alt="image" src="https://github.com/user-attachments/assets/f3cf90f1-a05d-4702-9589-94e75afb199e" />

and it defaults to `true` from the client side.

Reply:
I'm always down for making user preference names less confusing :D